task/result-handler.go

```diff
@@ -172,6 +185,45 @@ func pushResult(encryptedData []byte) error {
 	}
 }
 
+// retryPushResultInfinitely keeps retrying to push encrypted data to the callback URL.
+func retryPushResultInfinitely(encryptedData []byte) error {
+	retryDelay := 1 * time.Second
+	maxBackoff := 3 * time.Minute
+	backoffFactor := 2.0
+
+	for {
+		clientset, err := k8s.GetClientset()
+		if err != nil {
+			klog.Warningf("Cannot create clientset for handling Result Secret: %s", err)
+			return err
+		}
+		// we need to re-fetch callback URL since it will change when remote-runner restarts
+		secret, err := clientset.CoreV1().Secrets(os.Getenv(RunnerNamespace)).Get(context.Background(), os.Getenv(InputContextSecret), v1.GetOptions{})
+		if err != nil {
+			klog.Warningf("Cannot fetch Result Secret: %s", err)
+			return err
+		}
+		callbackUrl = string(secret.Data[InputContextSecretDataUrlKey])
+		decodedCallbackUrl, err := base64.StdEncoding.DecodeString(callbackUrl)
+		if err != nil {
+			klog.Warningf("Cannot decode Callback URL %s", err)
+			return err
+		}
+		url := string(decodedCallbackUrl)
+
+		_, httpError := http.Post(url, "application/json", bytes.NewReader(encryptedData))
+		if httpError == nil {
+			return nil
+		}
+
+		klog.Warningf("Cannot finish retried Callback request: %s. Retrying in %v...", httpError, retryDelay)
+		time.Sleep(retryDelay)
+
+		newDelay := time.Duration(float64(retryDelay) * backoffFactor)
+		retryDelay = time.Duration(math.Min(float64(newDelay), float64(maxBackoff)))
```

Comment:
maybe a better choice would be without going back and forth with float to duration casting.

Remove line 222 and change 223:
		`retryDelay = time.Duration(math.Min(float64(retryDelay) * backoffFactor, float64(maxBackoff)))`